ibm/service/power/resource_ibm_pi_network_address_group.go

```diff
@@ -0,0 +1,238 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceIBMPINetworkAddressGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPINetworkAddressGroupCreate,
+		ReadContext:   resourceIBMPINetworkAddressGroupRead,
+		UpdateContext: resourceIBMPINetworkAddressGroupUpdate,
+		DeleteContext: resourceIBMPINetworkAddressGroupDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(5 * time.Minute),
```

Comment:
Add create and update timeouts

Reply:
Not needed, here. The only one we are using is delete